### PR TITLE
Update span parser

### PR DESCRIPTION
**Pull request by xiao-lix**

### CODE REVIEW

File: AutoCollection/diagnostic-channel/SpanParser.ts [hunk 1]
```diff
@@ -17,22 +17,19 @@ export function spanToTelemetryContract(span: Span): (Contracts.DependencyTeleme
     const id = `|${span.spanContext().traceId}.${span.spanContext().spanId}.`;
     const duration = Math.round(span["_duration"][0] * 1e3 + span["_duration"][1] / 1e6);
     let peerAddress = span.attributes["peer.address"] ? span.attributes["peer.address"].toString() : "";
-    let component = span.attributes["component"] ? span.attributes["component"].toString() : "";
 
-    const isHttp: boolean = ((component).toUpperCase() === Constants.DependencyTypeName.Http) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
-    const isGrpc: boolean = (component).toLowerCase() === Constants.DependencyTypeName.Grpc;
+    const isHttp: boolean = (!!span.attributes[Constants.SpanAttribute.HttpStatusCode]) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
+    const isGrpc: boolean = (!!span.attributes[Constants.SpanAttribute.GrpcStatusCode]);
```
Comment (xiao-lix) [inline AutoCollection/diagnostic-channel/SpanParser.ts:22]: - to check if it's http, check span[HTTP_STATUS_CODE] [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L110)
- to check if it's grpc, check span[RPC_GRPC_STATUS_CODE] [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L115)

File: AutoCollection/diagnostic-channel/SpanParser.ts [hunk 2]
```diff
@@ -17,22 +17,19 @@ export function spanToTelemetryContract(span: Span): (Contracts.DependencyTeleme
...
         const url = new URL(span.attributes[Constants.SpanAttribute.HttpUrl].toString());
-        const host = span.attributes[Constants.SpanAttribute.HttpHost] || url.host;
-        const port = span.attributes[Constants.SpanAttribute.HttpPort] || url.port || null;
         const pathname = url.pathname || "/";
 
         // Translate to AI Dependency format
         const name = `${method} ${pathname}`;
         const dependencyTypeName = Constants.DependencyTypeName.Http;
-        const target = port ? `${host}:${port}`.toString() : host.toString();
+        const target = span.attributes[Constants.SpanAttribute.HttpUrl].toString() || undefined;
```
Comment (xiao-lix) [inline AutoCollection/diagnostic-channel/SpanParser.ts:32]: - for HTTP, target is `[new URL(span[HTTP_URL])].hostname` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L127)
- for GRPC, target is` span[GRPC_METHOD]` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L121)
- else, target is `span[HTTP_URL] | undefined` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L104)

File: Declarations/Constants.ts [hunk 3]
```diff
@@ -123,16 +124,15 @@ export const TelemetryTypeStringToQuickPulseDocumentType: {[key in Contracts.Tel
...
+    HttpStatusCode: conventions.SemanticAttributes.HTTP_STATUS_CODE,
+    HttpUrl: conventions.SemanticAttributes.HTTP_URL,
+    HttpUserAgent: conventions.SemanticAttributes.HTTP_USER_AGENT,
 
     // GRPC
-    GrpcMethod: "grpc.method",
-    GrpcService: "rpc.service", // rpc not grpc
+    GrpcStatusCode: conventions.SemanticAttributes.RPC_GRPC_STATUS_CODE,
+    GrpcMethod: conventions.SemanticAttributes.RPC_METHOD, // rpc not grpc
+    GrpcService: conventions.SemanticAttributes.RPC_SERVICE, // rpc not grpc
```
Comment (xiao-lix) [inline Declarations/Constants.ts:135]: All GRPC related constants [reference](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/monitor-opentelemetry-exporter/src/utils/constants/span/grpcAttributes.ts)

File: Declarations/Constants.ts [hunk 4]
```diff
@@ -123,16 +124,15 @@ export const TelemetryTypeStringToQuickPulseDocumentType: {[key in Contracts.Tel
...
-    HttpHost: "http.host",
-    HttpMethod: "http.method",
-    HttpPort: "http.port",
-    HttpStatusCode: "http.status_code",
-    HttpUrl: "http.url",
-    HttpUserAgent: "http.user_agent",
+    HttpMethod: conventions.SemanticAttributes.HTTP_METHOD,
+    HttpStatusCode: conventions.SemanticAttributes.HTTP_STATUS_CODE,
+    HttpUrl: conventions.SemanticAttributes.HTTP_URL,
+    HttpUserAgent: conventions.SemanticAttributes.HTTP_USER_AGENT,
```
Comment (xiao-lix) [inline Declarations/Constants.ts:130]: All HTTP related constants [reference](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/monitor-opentelemetry-exporter/src/utils/constants/span/httpAttributes.ts)

**Review by hectorhdzg — APPROVED** [review]
LGTM

**Pull request merged**